### DISCUSSION TITLE
Change resolution on indi_pylibcamera

### DISCUSSION THREAD

**Discussion by wjschmitt** [aaronwmorris/indi-allsky#1501] [General]
I'm running an RPi HQ camera using the indi_pylibcamera driver on the Pi, which sends images to indi_allsky running on a VM.   It looks like indi_pylibcamera runs the IMX477 at its highest resolution 4056x3040. by default.  I want to run at a lower resolution and higher framerate.  I haven't been able to figure out where I can do that in the indi_pylibcamera.ini.   Can I instead use the INDI custom config on the indi_allsky.  If so, can you help with the settings for the IMX477.  There isn't a section on the wiki for that camera.  Thanks.

**Comment by aaronwmorris** [comment]
I have not tried to adjust these options via indi_pylibcamera.  I am currently on vacation and I will play with these options when I get back later this week.

**Reply by wjschmitt** [reply]
No worries.  Have a nice vaca.  You deserve it.

**Comment by wjschmitt** [comment]
I asked this same question on the indi_pylibcamera page.  Ronald over there says there is no way to set resolution in pylibcamera.  He says however that I could use binning to do this on allsky end.  By default, BIN MODE (which I assume is the binning option) is set to 1 in Camera Config.  So setting it to 2 should halve the resolution.  I'll give this a try.

**Comment by wjschmitt** [comment]
On a related matter, is it even necessary to install indi_allsky software on the Pi if I will be using indi_pylibcamera to send camera data to the indi_allsky server?

**Reply by aaronwmorris** [reply]
I missed this question, sorry.

No, it is not necessary to have the full install of indi-allsky if you are running a remote indiserver with indi_pylibcamera.

**Comment by wjschmitt** [comment]
I set BIN MODE to 2 and now image resolution is 2024 x 1520, which is half.   So this is the way to adjust resolution.  Ronald said that the IMX477 only supports bin modes of 1 and 2.  Not sure what will happen if I set it to 3 or 4.

Now I have to reduce the size of the labels for the new resolution.

**Reply by aaronwmorris** [reply]
Makes sense.  You could also just use the scale option in indi-allsky to scale the image size down.  With libcamera, the binning modes do not provide any benefit to SNR like they would with most astronomy cameras.
